cmd/dagger/flags.go

```diff
@@ -423,7 +425,16 @@ func (v *secretValue) Get(ctx context.Context, c *dagger.Client, _ *dagger.Modul
 		plaintext = envPlaintext
 
 	case fileSecretSource:
-		filePlaintext, err := os.ReadFile(v.sourceVal)
+		sourceVal := v.sourceVal
+		// For unix, automatic expand tilde (~) to $HOME
+		if strings.HasPrefix(sourceVal, "~") {
+			usr, err := user.Current()
+			if err != nil {
+				return nil, err
+			}
+			sourceVal = strings.Replace(sourceVal, "~", usr.HomeDir, 1)
```

Comment:
What do you think of using `homedir.Expand` like we did in Go SDK?

Reply:
We used to use https://github.com/mitchellh/go-homedir instead of `os/user` for this.

Reply:
Maybe it's no longer an issue though:
- https://github.com/mitchellh/go-homedir/issues/34

Reply:
I see. Will take a look at it tomorrow. 🙏

Reply:
I ended up by using `os.UserHomeDir` to get home directory from shell environment but copy path detection from go-homedir instead. 🙇